src/openforms/js/components/admin/form_design/data/complete-form.js

```diff
@@ -62,6 +62,26 @@ const saveForm = async (state, csrftoken) => {
       // FIXME - should not be required in backend for form designer
       name: draft.form?.translations?.[DEFAULT_LANGUAGE]?.name,
       authenticationBackends: draft.selectedAuthPlugins,
+
+      // make sure empty str for PositiveIntegerField is converted to null.
```

Comment:
maybe create a helper function in this file for this?

```js
const normalizeLimit = (draft, field) => {
    const removalOptions = draft.form?.submissionsRemovalOptions;
    if (!removalOptions) return;
    const currentValue = removalOptions?.[field];
    if (currentValue === '') {
      removalOptions[field] = null;
    }
};
```

and then call it using:

```js
normalizeLimit(draft, 'successfulSubmissionsRemovalLimit');
// ...
```